### PR TITLE
feat(qiankun): 兼容 ssr

### PR DESCRIPTION
**Pull request by kuitos**
This change is [Reviewable](https://reviewable.io/reviews/umijs/plugins/223)

### CODE REVIEW

File: packages/plugin-qiankun/src/master/index.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ export default function(api: IApi, options: MasterOptions) {
     } = api;
     const masterHistoryType = history?.type || defaultHistoryType;
     const rootExports = `
-window.g_rootExports = ${
+    if (window) window.g_rootExports = ${
```
Comment (ycjcl868) [inline packages/plugin-qiankun/src/master/index.ts:114]:
```suggestion
    if (typeof window !== 'undefined') window.g_rootExports = ${
```

File: packages/plugin-qiankun/src/slave/runtimePlugin.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import qiankunRender from './lifecycles';
 
 export function rootContainer(container: HTMLElement) {
-  const value = (window as any).g_rootExports;
+  const value = window ? (window as any).g_rootExports : {};
```
Comment (ycjcl868) [inline packages/plugin-qiankun/src/slave/runtimePlugin.ts:5]:
```suggestion
  const value = typeof window !== 'undefined' ? (window as any).g_rootExports : {};
```

File: packages/plugin-qiankun/src/master/index.ts (hunk 1 above)
Comment (ycjcl868) [reply packages/plugin-qiankun/src/master/index.ts:114]: 括下

File: packages/plugin-qiankun/src/master/index.ts [hunk 3]
```diff
@@ -111,7 +111,7 @@ export default function(api: IApi, options: MasterOptions) {
     } = api;
     const masterHistoryType = history?.type || defaultHistoryType;
     const rootExports = `
-window.g_rootExports = ${
+    if (typeof window !== 'undefined') { window.g_rootExports = ${
       rootExportsFileExisted ? `require('@/rootExports')` : `{}`
     };
```
Comment (ycjcl868) [inline packages/plugin-qiankun/src/master/index.ts:116]: 少了个 `}`

Comment (ycjcl868) [inline packages/plugin-qiankun/src/master/index.ts:116]:
```suggestion
    };
}
```

Comment (kuitos) [reply packages/plugin-qiankun/src/master/index.ts]: done

**Pull request merged**